File: backend/src/main/kotlin/de/iteratec/konfi/domain/model/Conference.java
package de.iteratec.konfi.domain.model;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Conference {

    @Id
    private Long id;

    private String name;

    private int maxAttendees;

    private LocalDateTime startDate;

    private LocalDateTime endDate;

    @ElementCollection
    private List<Attendee> attendees = new ArrayList<>();

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getMaxAttendees() {
        return maxAttendees;
    }

    public void setMaxAttendees(int maxAttendees) {
        this.maxAttendees = maxAttendees;
    }

    public LocalDateTime getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDateTime startDate) {
        this.startDate = startDate;
    }

    public LocalDateTime getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDateTime endDate) {
        this.endDate = endDate;
    }

    public List<Attendee> getAttendees() {
        return attendees;
    }

    public void setAttendees(List<Attendee> attendees) {
        this.attendees = attendees;
    }

    public boolean collidesWith(Conference other) {
        return other.endDate.compareTo(startDate) >= 0 && other.startDate.compareTo(endDate) <= 0;
    }
}

File: backend/src/main/java/de/iteratec/konfi/domain/model/Conference.kt
package de.iteratec.konfi.domain.model

import java.time.LocalDateTime
import de.iteratec.konfi.domain.model.Attendee
import de.iteratec.konfi.domain.model.Conference
import java.util.ArrayList
import javax.persistence.*

@Entity
class Conference {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    var name: String? = null
    var maxAttendees = 0
    var startDate: LocalDateTime? = null
    var endDate: LocalDateTime? = null

    @ElementCollection
    var attendees: List<Attendee> = ArrayList()
    fun collidesWith(other: Conference): Boolean {
        return other.endDate!!.compareTo(startDate) >= 0 && other.startDate!!.compareTo(endDate) <= 0
    }
}
